ext/authx/README.md

```diff
@@ -127,3 +129,83 @@ $ curl 'https://demouser:demopass@example.org/civicrm/authx/id'
 
 The "AuthX: Authenticate to services with password" CiviCRM permission must
 also be granted for the role associated to the user.
+
+## Extending credential handling
```

Comment:
Woot. Documentation. :)

At some point, we'd probably want to move this to dev-docs. But I'm mostly just glad it's written. :smile: